### DISCUSSION TITLE
Tip for M1 Users on VRFCoordinator

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1288] [General]
Per usual when using the Mac M1, I'm (we?) are stuck using solidity ini the most recent version, for 0.8.12.  This becomes tricky in the smart lottery because the chainlink-mix VRFCoordinator is still using 0.6.0.  This one was actually EASIER to walk around than others have been in the past.  

First - the link for the VRF Coordinator you want to use for Mac M1 is [right here](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/tests/VRFCoordinatorV2TestHelper.sol)

Couple things you need to change.

One - change the pragma to `0.8.12`

Secondly there's an import of VRFCoordinatorV2.sol from a dev folder in the git hub and the 0.6.0 also brings in LinkTokenInterface, so I did that for good measure as well

```
import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
import "@chainlink/contracts/src/v0.8/LinkTokenInterface.sol";
```

I copied my version 0.8.12 MockAggregator from the Fund Me lesson and both compiled without a hitch which may be the first 'no issue' compile I've had.

Not sure this will help anyone else but wanted to share in case.

**Comment by cromewar** [comment]
Nice guide @JohnEMagee !!

**Reply by JohnEMagee** [reply]
One more important note (as I had to delete and reset cause I made a small mistake) - the contract that deploys is ***NOT*** VRFCoordinator, but VRFCoordinatorV2Mock - be sure to look in your build/contracts folder to make sure you're calling the right contract
